src/components/modules/dragNDrop.ts

```diff
@@ -13,39 +13,82 @@ export default class DragNDrop extends Module {
    */
   private isStartedAtEditor = false;
 
+  /**
+   * Listener identifiers
+   */
+  private listenerIds: string[] = [];
+
   /**
    * Bind events
    */
   public prepare(): void {
-    this.bindEvents();
+    if (!this.config.readOnly) {
+      this.enableModuleBindings();
+    }
   }
 
   /**
-   * Add drag events listeners to editor zone
+   * Toggle read-only state
    *
-   * @private
+   * if state is true:
+   *  - disable all drag-n-drop event handlers
+   *
+   * if state is false:
+   *  - restore drag-n-drop event handlers
+   *
+   * @param {boolean} readOnlyEnabled - "read only" state
    */
-  private bindEvents(): void {
-    this.Editor.Listeners.on(this.Editor.UI.nodes.holder, 'drop', this.processDrop, true);
+  public toggleReadOnly(readOnlyEnabled: boolean): void {
+    if (readOnlyEnabled) {
+      this.disableModuleBindings();
+    } else {
+      this.enableModuleBindings();
+    }
+  }
+
+  /**
+   * Add drag events listeners to editor zone
+   */
+  private enableModuleBindings(): void {
+    this.listenerIds.push(
+      this.Editor.Listeners.on(this.Editor.UI.nodes.holder, 'drop', async (dropEvent: DragEvent) => {
+        await this.processDrop(dropEvent);
+      }, true)
+    );
 
-    this.Editor.Listeners.on(this.Editor.UI.nodes.holder, 'dragstart', (dragEvent: DragEvent) => {
-      if (SelectionUtils.isAtEditor && !SelectionUtils.isCollapsed) {
-        this.isStartedAtEditor = true;
-      }
+    this.listenerIds.push(
+      this.Editor.Listeners.on(this.Editor.UI.nodes.holder, 'dragstart', () => {
+        this.processDragStart();
+      })
+    );
 
-      this.Editor.InlineToolbar.close();
-    });
+    /**
+     * Prevent default browser behavior to allow drop on non-contenteditable elements
+     */
+    this.listenerIds.push(
+      this.Editor.Listeners.on(this.Editor.UI.nodes.holder, 'dragover', (dragEvent: DragEvent) => {
```

Comment:
Better to rewrite handlers to arrow functions. Then arrow function is not needed here

Reply:
we can discuss it